src/pp.js

```diff
@@ -297,7 +311,14 @@ function print(w, doc) {
           break;
         case "if-break":
           if (mode === MODE_BREAK) {
-            cmds.push([ ind, MODE_BREAK, doc.contents ]);
+            if (doc.breakContents !== undefined) {
+              cmds.push([ ind, mode, doc.breakContents ]);              
+            }
+          }
+          if (mode === MODE_FLAT) {
+            if (doc.flatContents !== undefined) {
```

Comment:
Yeah, I think @rattrayalex is right. It makes for cleaner code and would allow `null` which I guess is OK so let's convert it to that instead.